_posts/2019-07-17-#15169.md

```diff
@@ -7,7 +7,30 @@ title: "#15169 Parallelize CheckInputs() in AcceptToMemoryPool()"
 
 ## Notes
 
+* The [CheckInputs()](https://github.com/bitcoin/bitcoin/blob/8f9725c83f1da5364354d2820696c84d542d3d37/src/validation.cpp#L1234) function is called when checking the inputs to a transaction. Each input has a script which is verified. That verification can be run:
+    - sequentially and synchronously, in which case `CheckInputs()` will return `true` if all scripts pass verification or pass back the error for the first script that failed verification; or
+    - in parallel, in which case `CheckInputs()` will emplace the script check onto a queue for future verification.
+* Previously, `CheckInputs()` would always check inputs sequentially when accepting a transaction to the mempool. Transaction inputs could be checked in parallel during block validation. Note the [`vChecks` argument when calling `CheckInputs()` from `ConnectBlock()`](https://github.com/bitcoin/bitcoin/blob/8f9725c83f1da5364354d2820696c84d542d3d37/src/validation.cpp#L1887).
+* This PR enables the input checks to be performed in parallel when a transaction is entering the mempool. 
+* Currently, `CheckInputs()` will validate the inputs of a new transaction in sequence, and is called updating the mempool with a new transaction. 
+* This PR replaces calls to `CheckInputs()` with `RunCheckInputsMaybeParallel()`, which will push the input checks to an existing CCheckQueue worker queue.
+  * `CheckInputs`: https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp#L1234
+  * Checkqueue (Currently used for Block validation): https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp#L1528 
+  * Script check worker loop (Currently used for Block validation): https://github.com/bitcoin/bitcoin/blob/master/src/checkqueue.h#L66 
```

Comment:
s/Block/block/ (this item seems unfinished)

Reply:
Not sure what you mean?

Reply:
I meant that "block" didn't need to be capitalized, but it's a nit not worth fixing unless there is more you want to change.